src/cookbooks/bigquery.md

```diff
@@ -0,0 +1,148 @@
+- [Databricks](https://sso.mozilla.com/databricks) via BigQuery Spark Connector which uses [BigQuery Storage API](https://cloud.google.com/bigquery/docs/reference/storage/)
+
+For access to BigQuery via GCP Console and API please file a bug [here](https://bugzilla.mozilla.org/enter_bug.cgi?assigned_to=jthomas%40mozilla.com&bug_file_loc=https%3A%2F%2Fmana.mozilla.org%2Fwiki%2Fx%2FiIPeB&bug_ignored=0&bug_severity=normal&bug_status=NEW&bug_type=task&cf_fx_iteration=---&cf_fx_points=---&comment=Please%20grant%20me%20access%20to%20the%20BigQuery%20GCP%20console%20and%20API%20Access.%20I%20work%20on%20%3Cteam%3E.%0D%0A%0D%0AMy%20mozilla.com%20ldap%20login%20is%20%3Cyour%20ldap%20login%3E%40mozilla.com.&component=Operations&contenttypemethod=list&contenttypeselection=text%2Fplain&defined_groups=1&flag_type-4=X&flag_type-607=X&flag_type-800=X&flag_type-803=X&flag_type-936=X&form_name=enter_bug&maketemplate=Remember%20values%20as%20bookmarkable%20template&op_sys=Unspecified&priority=--&product=Data%20Platform%20and%20Tools&qa_contact=jthomas%40mozilla.com&rep_platform=Unspecified&short_desc=BigQuery%20GCP%20Console%20and%20API%20Access%20for%20%3Cyour%20ldap%20login%3E%40mozilla.com&target_milestone=---&version=unspecified). As part of this request we will add you to the appropriate Google Groups and provision a GCP Service Account.
+
+
+# Query Optimizations
+
+To improve query performance and minimize the cost associated with using BigQuery please see the following query optimizations:
+
+- Avoid `SELECT *` and query only the columns you need
+    - Using `SELECT *` is the most expensive way to query data. When you use `SELECT *` _BigQuery does a full scan of every column in the table._
+    - Applying a `LIMIT` clause to a `SELECT *` query does not affect of data read. You are billed for reading all bytes in the entire table.
+    - If you are experimenting with data or exploring data, use one of the [data preview options](https://cloud.google.com/bigquery/docs/best-practices-costs#preview-data) instead of `SELECT *`.
+        - Preview support is coming soon to BigQuery datasources in [Redash](https://sql.telemetry.mozilla.org/)
+- Limit the amount of data scanned using a date partition filter
+    - Tables that are larger than 1TB will require that you provide a date partition filter as part of the query.
+    - You will receive an error if you attempt to query a table that requires a partition filter.
+        - `Cannot query over table 'moz-fx-data-derived-datasets.telemetry.main_summary_v4' without a filter over column(s) 'submission_date_s3' that can be used for partition elimination`
+- Reduce data before using a JOIN
+    - Trim the data as early in the query as possible, before the query performs a JOIN. If you reduce data early in the processing cycle, shuffling and other complex operations only execute on the data that you need.
```

Comment:
I'm not sure of the exact audience here, but some examples of how to reduce(decrease/lessen) a query's output set prior to joining may be helpful here. Also 'reduce' could also be confused with map/reduce.

Reply:
I think this would vary based on what data is being queried. Do you have suggestions on a generic example?

Reply:
Ah I wasn't entirely was sure about what you meant by reducing data, so was asking for examples. https://cloud.google.com/bigquery/docs/best-practices-performance-compute suggests using the largest table first in a join.

I'm assuming you meant using intermediate tables/views or sub queries with multiple where clauses as a way of decreasing sides of a join, prior to the join itself.

Reply:
> Ah I wasn't entirely was sure about what you meant by reducing data, so was asking for examples. https://cloud.google.com/bigquery/docs/best-practices-performance-compute suggests using the largest table first in a join.

The optimization you've mentioned is _yet another_ optimization that users can do :). 

> I'm assuming you meant using intermediate tables/views or sub queries with multiple where clauses as a way of decreasing sides of a join, prior to the join itself.

Yes this. I am just going to add that instead of providing an actual example.